TPL/force-app/main/default/classes/HCCostCaseDAO.cls

```diff
@@ -1,83 +1,187 @@
 public with sharing class HCCostCaseDAO {
-    public static List<Healthcare_Cost__c> getHCCostCaseRecords(String caseId, String filterValue, Id recordTypeId, Integer startItemNumber, Integer pageSize, String fields){
-        Integer itemsToRemove = startItemNumber - 2;
-        System.debug(itemsToRemove);
-        list<Healthcare_Cost__c> hccList = new list<Healthcare_Cost__c>();
-        String dateId = '';
-        String query = '';
-        String sourceSystemId = null;
-        Boolean costValue = false;
-        try{
-           
-            if(filterValue == Label.TPL_Manual_Records){
-                if(itemsToRemove > 2000)
+public static List<Healthcare_Cost__c> getHCCostCaseRecords(String caseId, String filterValue, Id recordTypeId, Integer startItemNumber, Integer pageSize, String fields, String sortOrder){
+Integer itemsToRemove = startItemNumber - 2;
+
+list<Healthcare_Cost__c> hccList = new list<Healthcare_Cost__c>();
+String dateId = '';
+String query = '';
+String sourceSystemId = null;
+Boolean costValue = false;
+try{
+    if(sortOrder == null || sortOrder == ''){
+        sortOrder = 'asc';
+    }
+
+    if(sortOrder == Label.TPL_Descending){
```

Comment:
It was determined that merging these processes under this branch would create more code that leaving them separate due to building complex strings for the many queries for stepping through the dataset.